components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt

```diff
@@ -192,6 +192,19 @@ class GeckoEngine(
         })
     }
 
+    /**
+     * See [Engine.updateWebExtension].
+     */
+    override fun updateWebExtension(
+        extension: WebExtension,
+        onSuccess: (WebExtension?) -> Unit,
+        onError: (String, Throwable) -> Unit
+    ) {
```

Comment:
Are`onSuccess` or `onError` callbacks  going to be called when GeckoView notifies us that the updated extension have added extra permissions and user interaction is required?

Reply:
@Amejia481 I didn't think so, no. There'll be a `PromptDelegate` which we will set using `setPromptDelegate` on the `WebExtensionController`. So, `onSuccess` will be called once the exension is updated, `onError` in case an actual error occurred, but we can double-check.

Reply:
Verified. Only  `onUpdatePrompt` will be called if the permissions changed.